rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/PostgresqlUndoLogManager.java

```diff
@@ -0,0 +1,271 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.rm.datasource.undo.postgresql;
+
+import com.alibaba.druid.util.JdbcConstants;
+import io.seata.core.constants.ClientTableColumnsName;
+import io.seata.core.exception.BranchTransactionException;
+import io.seata.core.exception.TransactionException;
+import io.seata.rm.datasource.DataSourceProxy;
+import io.seata.rm.datasource.sql.struct.TableMeta;
+import io.seata.rm.datasource.sql.struct.TableMetaCacheFactory;
+import io.seata.rm.datasource.undo.AbstractUndoExecutor;
+import io.seata.rm.datasource.undo.AbstractUndoLogManager;
+import io.seata.rm.datasource.undo.BranchUndoLog;
+import io.seata.rm.datasource.undo.SQLUndoLog;
+import io.seata.rm.datasource.undo.UndoExecutorFactory;
+import io.seata.rm.datasource.undo.UndoLogConstants;
+import io.seata.rm.datasource.undo.UndoLogParser;
+import io.seata.rm.datasource.undo.UndoLogParserFactory;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.seata.core.exception.TransactionExceptionCode.BranchRollbackFailed_Retriable;
+
+/**
+ * @author japsercloud
+ */
+public class PostgresqlUndoLogManager extends AbstractUndoLogManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresqlUndoLogManager.class);
+
+    private static final String INSERT_UNDO_LOG_SQL = "INSERT INTO " + UNDO_LOG_TABLE_NAME + "\n" +
+        "\t(id,branch_id, xid,context, rollback_info, log_status, log_created, log_modified)\n" +
+        "VALUES (nextval('undo_log_seq'),?, ?,?, ?, ?, now(), now())";
+
+    private static final String DELETE_UNDO_LOG_BY_CREATE_SQL = "DELETE FROM " + UNDO_LOG_TABLE_NAME +
+        " WHERE log_created <= ? LIMIT ?";
+
+    @Override
+    public String getDbType() {
+        return JdbcConstants.POSTGRESQL;
+    }
+
+    /**
+     * Undo.
+     *
+     * @param dataSourceProxy the data source proxy
+     * @param xid the xid
+     * @param branchId the branch id
+     * @throws TransactionException the transaction exception
+     */
+    @Override
+    public void undo(DataSourceProxy dataSourceProxy, String xid, long branchId) throws TransactionException {
```

Comment:
do not need override parent's undo method, only implements parent's abstract method.